packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -564,298 +705,303 @@ function condenseErrorsAndFilterSiblings<T>(
 		// and look sth like this:
 		// Missing either keys ("A", "B" and "C"), ("A", and "D"), or "F"
 		errors.push(
-			...validDefinitions.map(d => ({
-				kind: 'some_missing_keys' as 'some_missing_keys',
-				node: d.node.runtimeNode.node,
-			})),
+			...validDefinitions
+				.flatMap(d =>
+					d.errors
+						.filter(e =>
+							e.kind === 'missing_key' &&
+							!alwaysMissing.some(oe =>
+								oe.node.originalNode === e.node.originalNode
+							)
+						)
+				)
+				.map(e => e.node as RuntimeNode<T>)
+				.concat(
+					validDefinitions.flatMap(d => d.errors).filter(e =>
+						e.kind === 'some_missing_keys'
+					).map(e => e.node as RuntimeNode<T>),
+				)
+				.filter((v, i, arr) =>
+					arr.findIndex(o => o.originalNode === v.originalNode) === i
+				)
+				.map(n => ({
+					kind: 'some_missing_keys' as 'some_missing_keys',
+					node: n,
+				})),
 		)
 	}
 
 	// TODO handle list length range and value range errors (merge ranges, could be multiple possible distinct ranges)
 
 	return {
-		siblings: validDefinitions.map(d => d.node),
+		definitions: validDefinitions.map(d => d.definition),
 		condensedErrors: errors,
 	}
 }
 
-function check<T>(
-	node: CheckerTreeEntryNode<T>,
+interface ValidDefintionResult<T> {
+	errors: McdocCheckerError<T>[]
+	childDefinitions: (McdocType | undefined)[]
+}
+
+function checkShallowly<T>(
+	runtimeNode: RuntimeNode<T>,
+	simplifiedInferred: SimplifiedMcdocTypeNoUnion,
+	children: RuntimeUnion<T>[],
+	typeDef: SimplifiedMcdocTypeNoUnion,
 	options: McdocCheckerOptions<T>,
-) {
-	if (node.typeDef === undefined) {
-		return
-	}
-	const parents: RuntimeUnion<T>[] = []
-	let current: CheckerTreeEntryNode<T> | undefined = node
-	while (current) {
-		const possibleValues = current.possibleRuntimeValues.map(v => v.node)
-		parents.unshift(
-			current.runtimeKey
-				? { key: current.runtimeKey, possibleValues }
-				: possibleValues,
-		)
-		current = current.parent?.runtimeNode.entryNode
+): ValidDefintionResult<T> {
+	const typeDefValueType = getValueType(typeDef)
+	const runtimeValueType = getValueType(simplifiedInferred)
+
+	if (
+		(typeDef.kind !== 'any' && typeDef.kind !== 'unsafe' &&
+			runtimeValueType.kind !== typeDefValueType.kind &&
+			!options.isEquivalent(runtimeValueType, typeDefValueType)) ||
+		(typeDef.kind === 'literal' &&
+			(simplifiedInferred.kind !== 'literal' ||
+				typeDef.value.value !== simplifiedInferred.value.value)) ||
+		// TODO handle enum field attributes
+		(typeDef.kind === 'enum' &&
+			(simplifiedInferred.kind !== 'literal' ||
+				!typeDef.values.some(v =>
+					v.value === simplifiedInferred.value.value
+				)))
+	) {
+		return {
+			childDefinitions: Array(children.length).fill(undefined),
```

Comment:
What is the point of this? Since there is a type mismatch, why not return an empty list of definitions?